frontend/javascripts/admin/voxelytics/utils.ts

```diff
@@ -63,3 +65,29 @@ export async function copyToClipboad(text: string) {
   await navigator.clipboard.writeText(text);
   message.success("Copied to clipboard");
 }
+
+export async function loadAllLogs(
+  runId: string,
+  taskName: string | null,
+  minLevel: LOG_LEVELS,
+  startTime: Date,
+  endTime: Date,
+  batchSize: number = 5000,
+): Promise<Array<VoxelyticsLogLine>> {
+  let buffer: Array<VoxelyticsLogLine> = [];
+  let currentEndTime = endTime;
+  while (true) {
+    const batch = await getVoxelyticsLogs(
+      runId,
+      taskName,
+      minLevel,
+      startTime,
+      currentEndTime,
+      batchSize,
+    );
+    if (batch.length === 0) break;
+    buffer = batch.concat(buffer);
+    currentEndTime = new Date(batch[batch.length - 1].timestamp - 1);
```

Comment:
Shouldn't `startTime` be updated here as well?

Reply:
I don't think that is necessary. The `startTime` stays at the minimum `beginTime ?? new Date(0)`, because the logs are fetch in backward order from the `currentEndTime`.

Reply:
Ah I see, this works because of the limit when requesting the logs. Maybe the `getVoxelyticsLogs` function should contain a comment indicating that the limit is counted from `endTime`, so that the latest logs are fetched first, to avoid such confusion.

Reply:
I added a comment :+1: